Cubical/Algebra/BooleanRing/Base.agda

```diff
@@ -0,0 +1,246 @@
+{-# OPTIONS --safe #-}
+module Cubical.Algebra.BooleanRing.Base where
+
+open import Cubical.Foundations.Prelude hiding (_∧_;_∨_)
+open import Cubical.Foundations.Structure
+open import Cubical.Data.Empty as ⊥
+open import Cubical.Algebra.Ring
+open import Cubical.Algebra.AbGroup.Base
+open import Cubical.Algebra.CommRing
+open import Cubical.Tactics.CommRingSolver
+
+
+private
+  variable
+   ℓ ℓ' : Level
+
+record IsBooleanRing {B : Type ℓ}
+              (𝟘 𝟙 : B) (_+_ _·_ : B → B → B) (-_ : B → B) : Type ℓ where
+
```

Comment:
You might want to add "no-eta-equality" here, since these fields will all consist of propositions only, so their definition should never be used anyway.